Source/Core/Common/NandPaths.cpp

```diff
@@ -151,4 +152,12 @@ std::string UnescapeFileName(const std::string& filename)
 
   return result;
 }
+
+bool IsFileNameSafe(const std::string& filename)
+{
+  return !std::all_of(filename.begin(), filename.end(), [](char c) { return c == '.'; }) &&
+         std::none_of(filename.begin(), filename.end(), [](char c) {
+           return (c >= 0 && c <= 0x1F) || illegal_chars.find(c) != illegal_chars.end();
+         });
```

Comment:
This logic appears up above too. If you put it into a function IsIllegalCharacter() that would reduce duplication and make this statement simpler.